core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -847,7 +846,12 @@ void handleJoinRequest(final DiscoveryNode node, final MembershipAction.JoinCall
 
             // validate the join request, will throw a failure if it fails, which will get back to the
             // node calling the join request
-            membership.sendValidateJoinRequestBlocking(node, joinTimeout);
+            try {
+                membership.sendValidateJoinRequestBlocking(node, state, joinTimeout);
+            } catch (Throwable e) {
+                callback.onFailure(new IllegalStateException("Can't handle join request from node [" + node + "], probable cause missing plugins", ExceptionsHelper.unwrapCause(e)));
```

Comment:
can we change the message to what it is? "failure why sending a validation request to node" . The plugin part can better go in the type resolution (i.e., lookupPrototypeSafe ) ?